nym-data-observatory/src/modules/wasm.rs

```
@@ -0,0 +1,117 @@
+// Copyright 2025 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::db::DbPool;
+use crate::db::queries::wasm::insert_wasm_execute;
+use async_trait::async_trait;
+use cosmrs::proto::cosmwasm::wasm::v1::MsgExecuteContract;
+use cosmrs::proto::prost::Message;
+use nym_validator_client::nyxd::{Any, Name};
+use nyxd_scraper_psql::models::DbCoin;
+use nyxd_scraper_psql::{
+    MsgModule, NyxdScraperTransaction, ParsedTransactionResponse, ScraperError,
+};
+use serde_json::Value;
+use time::{OffsetDateTime, PrimitiveDateTime};
+use tracing::{error, trace};
+use utoipa::r#gen::serde_json;
+
+pub struct WasmModule {
+    connection_pool: DbPool,
+}
+
+impl WasmModule {
+    pub fn new(connection_pool: DbPool) -> Self {
+        WasmModule { connection_pool }
+    }
+}
+
+#[async_trait]
+impl MsgModule for WasmModule {
+    fn type_url(&self) -> String {
+        MsgExecuteContract::type_url()
+    }
+
+    async fn handle_msg(
+        &mut self,
+        index: usize,
+        msg: &Any,
+        tx: &ParsedTransactionResponse,
+        _storage_tx: &mut dyn NyxdScraperTransaction,
+    ) -> Result<(), ScraperError> {
+        let message = serde_json::to_value(tx.parsed_messages.get(&index)).unwrap_or_default();
```

I think rather tuan doing so many `unwrap_or_default()` you should return from this method instead otherwise we might run into some weird issues down the line (like the silent serialisation failures in node status api). while not ideal, maybe return `Ok(())` instead?

---

it is ok to have `null` as the default, because we can always reprocess blocks later

you are right about the silent failures though, however, this is what bdjuno does, so going to keep this behaviour going, and you can tell me "I told you so" later 😬